### DISCUSSION TITLE
不使用stomp协议的skynet rabbitmq模块

### DISCUSSION THREAD

**Discussion by Tianze-Chen** [cloudwu/skynet#1189] [Show and tell]
多数固有的rabbitmq队列并不支持stomp协议，希望能合入一个ampq协议的rabbitmq工具。
建议参考：
[https://github.com/gsdenys/amqp-client](url)
经测试、修改后可以正常使用，已知消息少的时候会出现异常、需要写入权限才能消费

**Comment by cloudwu** [comment]
我认为不宜再增加现有 skynet 仓库的维护负担。任何库的引入，都意味着有责任保证它没有问题。所以，建议把链接放在 wiki 里面。

**Comment by cloudfreexiao** [comment]
我写过个 可以参考下纯 lua 实现 如果需要修改 应该问题也不大 能用   额 我好像看错了问题 )
https://github.com/cloudfreexiao/skynet_rabbitmq

**Comment by cloudwu** [comment]
上面那个库我的个人意见是：应该把位操作、64bit 整数运算、解包函数这些，用新的 lua 内置语法重新实现以下。

**Comment by Tianze-Chen** [comment]
> 我写过个 可以参考下纯 lua 实现 如果需要修改 应该问题也不大 能用 额 我好像看错了问题 )
> https://github.com/cloudfreexiao/skynet_rabbitmq

哈哈 有拜读过这个实现，现有的队列不支持stomp，撕逼无果之后选择换了个方案

**Comment by Tianze-Chen** [comment]
> 上面那个库我的个人意见是：应该把位操作、64bit 整数运算、解包函数这些，用新的 lua 内置语法重新实现以下。

我加了一个bit.lua 简单跑通了这个解决方案，没有做比较好的优化，没好意思上传。稍后我试试放wiki上

**Comment by cloudfreexiao** [comment]
你贴的 链接 可以拿来改改 就是 代码多了点

**Comment by cloudwu** [comment]
1. string.unpack 可以替代掉反复的 string.byte 解码。
2. bit 库现在已经可以有内建的 bit 运算解决。
3. 现在整数已经支持 64bit 不再需要刻意考虑，但整数除法应该考虑使用 // 。
